mergesort.rb

```diff
@@ -1,24 +1,43 @@
 def mergesort(a)
     # if the array size is 0|1 then the list is considered sorted, return sorted array
     # split the list into two halves
+    if a.length == 1 || a.length == 0
+      return a
+    end
+    left, right = split_array(a) #if odd I put one less element on left
     # merge sort each half
+    left = mergesort(left)
+    right = mergesort(right)
     # combine the sorted halves
+    combine(left, right)
 end
 
 def split_array(a)
     # find the middle index
+    mid_index = a.length / 2
     # split the array in half - take a look at take and drop in the ruby docs, or use ranges
+    left = a.take(mid_index)
+    right = a.drop(mid_index)
     # return left and right halves of array as separate arrays - did you know a method can return multiple arrays? look it up
+    return left, right
 end
 
 # precondition: a and b are sorted
 def combine(a, b)
     # create a results array
-    # counters pointing to the index of the smallest elements in each array
+    result = []
+
     # check that we have elements to compare
-        # push the smaller element onto the result array
     # if there are elements left over in a, move them to result
     # if there are elements left over in b, move them to result
+    while a.size > 0 && b.size > 0
+      if a.first <= b.first
+        result << a.shift
+      else
+        result << b.shift
+      end
+    end
+    result.concat(a).concat(b)
 end
 
```

Comment:
Good solution all-around. Your combine method in particular is simple and readable. Good style points!